src/snapshot.ts

```diff
@@ -0,0 +1,346 @@
+   */
+  value(itemValue: (item: Item) => number): ItemResult {
+    // TODO: add garbo specific pricing (sugar equipment for synth, etc.)
+
+    const meat = Math.floor(this.meat);
+    const itemDetails = [...this.items.entries()].map(([item, quantity]) => {
+      return { item, quantity, value: itemValue(item) * quantity };
+    });
+    const items = Math.floor(sumNumbers(itemDetails.map((detail) => detail.value)));
+
+    return { meat, items, total: meat + items, itemDetails };
+  }
+
+  /**
+   * Subtract the contents of another snapshot from this one, removing any items that have a resulting quantity of 0
+   *  (this will ignore elements in b but not in a)
+   * @param other the snapshot from which to pull values to remove from this snapshot
+   * @returns a new snapshot representing the difference between this snapshot and the other snapshot
+   */
+  diff(other: Snapshot): Snapshot {
+    return new Snapshot(
+      this.meat - other.meat,
+      inventoryOperation(this.items, other.items, (a: number, b: number) => a - b, false)
+    );
+  }
+  /**
+   * Subtract the contents of snasphot b from snapshot a, removing any items that have a resulting quantity of 0
+   *  (this will ignore elements in b but not in a)
+   * @param a the snapshot from which to subtract elements
+   * @param b the snapshot from which to add elements
+   * @returns a new snapshot representing the difference between a and b
+   */
+  static diff(a: Snapshot, b: Snapshot): Snapshot {
+    return a.diff(b);
+  }
+
+  /**
+   * Generate a new Snapshot combining multiple snapshots together
+   * @param other the snapshot from which to add elements to this set
+   * @returns a new snapshot representing the addition of other to this
+   */
+  add(other: Snapshot): Snapshot {
+    return new Snapshot(
+      this.meat + other.meat,
+      inventoryOperation(this.items, other.items, (a: number, b: number) => a + b, true)
+    );
+  }
+
+  /**
+   * Combine the contents of snapshots
+   * @param snapshots the set of snapshots to combine together
+   * @returns a new snapshot representing the difference between a and b
+   */
+  static add(...snapshots: Snapshot[]): Snapshot {
+    return snapshots.reduce((previousSnapshot, currentSnapshot) =>
+      previousSnapshot.add(currentSnapshot)
+    );
+  }
+
+  /**
+   * Export this snapshot to a file in the data/ directory. Conventionally this file should end in ".json"
+   * @param filename The file into which to export
+   */
+  toFile(filename: string): void {
+    const val = {
+      meat: this.meat,
+      items: Object.fromEntries(this.items),
+    };
+    bufferToFile(JSON.stringify(val), filename);
+  }
+
+  /**
+   * Import a snapshot from a file in the data/ directory. Conventionally the file should end in ".json"
+   * @param filename The file from which to import
+   * @returns the Snapshot represented by the file
+   */
+  static fromFile(filename: string): Snapshot {
```

Comment:
is there something we can do other than throw/abort if we put in an invalid file?

Reply:
I am made...nervous isn't the right word, but wiggly, by the idea of having a function that acts on _arbitrary_ data files.

Reply:
Mafia only lets you read from the data/ directory, so it can only read files from there

Reply:
yeah, but I guess I mean what happens if somebody uses this function with a filename that points at something else? you'll get a big ol' error. Are we fine getting a big ol' error, or do we want to return null, or is there a mechanism where we can makethis only get pointed at files that we write?